### DISCUSSION TITLE
Partial file hydration?

### DISCUSSION THREAD

**Discussion by FeralSquid** [microsoft/ProjFS-Managed-API#72] [Q&A] [answered]
Hello,

Reading the documentation for GetFileDataCallback (PRJ_GET_FILE_DATA_CB), it sounds like this can / should be called requesting bytes for only certain ranges of the file that are being requested, but so far I've been unable to have this callback invoked with anything but the entire file being requested (byteOffset=0, length=fileSize).

Among other things, I've tried reading a byte from the middle of a file using a file handle created with FILE_FLAG_NO_BUFFERING, etc.

Is there a way to support/enable/trigger sub-full-file hydration?  We have a use case where it would be very advantageous to support this (many multi-GB files that always have their first ~120KB read but the rest of the file is read sparsely, if ever).

Thanks

**Comment by cgallred** [comment, marked as answer]
The API is designed to support partial hydration, but the platform doesn't do it right now. This is one of two features (the other being directory rename) that we want to support but have not yet implemented.

**Reply by turbo-grandma** [reply]
Hi, question from 2024 - is partial hydration feature getting implemented by any chance?
